File: python-sensitiveinfodisclosure/secure8.py
import json
import anthropic

client = anthropic.Anthropic()

with open("config.json") as f:
    config = json.load(f)


def get_deployment_help(issue):
    env_name = config["deployment"]["environment"]
    region = config["deployment"]["region"]
    response = client.messages.create(
        model="claude-3-5-sonnet-20241022",
        max_tokens=512,
        messages=[
            {"role": "user", "content": f"Help diagnose this issue in {env_name} ({region}): {issue}"}
        ],
    )
    return response.content[0].text


if __name__ == "__main__":
    print(get_deployment_help("health check endpoint returning 503"))

File: go-sensitiveinfodisclosure/secure8.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"net/http"
	"os"

	"github.com/anthropics/anthropic-sdk-go"
)

type AppConfig struct {
	SecretKey   string `json:"secret_key"`
	Environment string `json:"environment"`
	Region      string `json:"region"`
}

var client = anthropic.NewClient()

func getDeploymentHelp(ctx context.Context, cfg AppConfig, req *http.Request, issue string) (string, error) {
	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", cfg.SecretKey))

	msg, err := client.Messages.New(ctx, anthropic.MessageNewParams{
		Model:     anthropic.ModelClaude3_5Sonnet20241022,
		MaxTokens: 512,
		Messages: []anthropic.MessageParam{
			{Role: "user", Content: fmt.Sprintf("Help diagnose this issue in %s (%s): %s", cfg.Environment, cfg.Region, issue)},
		},
	})
	if err != nil {
		return "", err
	}
	return msg.Content[0].Text, nil
}

func main() {
	configBytes, _ := os.ReadFile("config.json")
	var cfg AppConfig
	json.Unmarshal(configBytes, &cfg)
	result, _ := getDeploymentHelp(context.Background(), cfg, &http.Request{Header: http.Header{}}, "health check returning 503")
	fmt.Println(result)
}
